### PR TITLE
Added the bot-notification changes for attachment

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
#### Summary
This PR will send the board bot notification to all the followers for the attachment changes.

#### Ticket Link
Fixes https://github.com/mattermost/focalboard/issues/4276

### CODE REVIEW

File: server/services/notify/notifysubscriptions/diff2slackattachments.go [hunk 1]
```diff
@@ -264,9 +267,34 @@ func appendCommentChanges(fields []*mm_model.SlackAttachmentField, cardDiff *Dif
 	return fields
 }
 
+func appendAttachmentChanges(fields []*mm_model.SlackAttachmentField, cardDiff *Diff) []*mm_model.SlackAttachmentField {
+	for _, child := range cardDiff.Diffs {
+		if child.BlockType == model.TypeAttachment {
+			var format string
+			var msg string
+			if child.NewBlock != nil && child.OldBlock == nil {
+				format = "Added the attachment by name: **`%s`**"
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply server/services/notify/notifysubscriptions/diff2slackattachments.go:276]: Sure.

**Review by sbishel — APPROVED** [review]

File: server/services/notify/notifysubscriptions/diff2slackattachments.go [hunk 2]
```diff
@@ -264,9 +267,34 @@ func appendCommentChanges(fields []*mm_model.SlackAttachmentField, cardDiff *Dif
...
+				msg = child.NewBlock.Title
+			} else {
+				format = "Removed ~~`%s`~~ attachment"
+				msg = stripNewlines(child.OldBlock.Title)
+			}
+
+			if format != "" {
+				fields = append(fields, &mm_model.SlackAttachmentField{
+					Short: false,
+					Title: "Attachment by " + makeAuthorsList(child.Authors, "unknow_user"), // TODO:  localize this when server has i18n
```
Comment (justinegeffen) [inline server/services/notify/notifysubscriptions/diff2slackattachments.go:286]:
```suggestion
					Title: "Changed by " + makeAuthorsList(child.Authors, "unknow_user"), // TODO:  localize this when server has i18n
```

File: server/services/notify/notifysubscriptions/diff2slackattachments.go (hunk 1 above)
Comment (justinegeffen) [inline server/services/notify/notifysubscriptions/diff2slackattachments.go:276]:
```suggestion
				format = "Added an attachment: **`%s`**"
```

[the remark this answers is not in the document]
Comment (justinegeffen) [reply server/services/notify/notifysubscriptions/diff2slackattachments.go:276]: Thanks, @wiggin77!

I've provided suggestions for changes to the header and the body copy. Is it possible to generate an example with that content so I can review?

File: server/services/notify/notifysubscriptions/diff2slackattachments.go [hunk 3]
```diff
@@ -264,9 +267,34 @@ func appendCommentChanges(fields []*mm_model.SlackAttachmentField, cardDiff *Dif
...
+				msg = child.NewBlock.Title
+			} else {
+				format = "Removed ~~`%s`~~ attachment"
+				msg = stripNewlines(child.OldBlock.Title)
+			}
+
+			if format != "" {
+				fields = append(fields, &mm_model.SlackAttachmentField{
+					Short: false,
+					Title: "Changed by " + makeAuthorsList(child.Authors, "unknow_user"), // TODO:  localize this when server has i18n
```
Comment (justinegeffen) [inline server/services/notify/notifysubscriptions/diff2slackattachments.go:286]: Should this be "unknown_user"?

Comment (Rajat-Dabade) [reply server/services/notify/notifysubscriptions/diff2slackattachments.go:286]: Yup, it should be.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   Great work @Rajat-Dabade

File: server/services/notify/notifysubscriptions/diff2slackattachments.go (hunk 3 above)
Comment (justinegeffen) [inline server/services/notify/notifysubscriptions/diff2slackattachments.go:286]:
```suggestion
					Title: "Changed by " + makeAuthorsList(child.Authors, "unknown_user"), // TODO:  localize this when server has i18n
```

Comment (Rajat-Dabade) [reply server/services/notify/notifysubscriptions/diff2slackattachments.go]: Got it.

**Comment by ogi-m** [comment]
@Rajat-Dabade I see a notification when adding an attachment, but no notification when an attachment is removed.

**Comment by Rajat-Dabade** [comment]
As the notification for removing the attachment was complex (due to architectural changes in the database i.e. hard delete of the child component) I and @wiggin77 decided to do it as a part of another PR. I will be creating a ticket for it. As of now, we can merge this PR. 

cc: @sbishel @ogi-m

**Review by ogi-m — APPROVED** [review]

**Pull request merged**